sdk/src/test/tutorial/java/com/finbourne/lusid/tutorials/ibor/ReferencePortfolio.java

```diff
@@ -0,0 +1,149 @@
+        String F40PortfolioCode = "F40J_ReferencePortfolioCode";
+        String F40PortfolioName = "F40J_Reference Portfolio name";
+        OffsetDateTime effectiveDate = OffsetDateTime.of(2010, 1, 1, 0, 0, 0, 0, ZoneOffset.UTC);
+        List<Double> weights = Arrays.asList(10.0, 20.0, 30.0, 15.0, 25.0);
+
+        Map<String, Double> instrumentWeights = IntStream.range(0, instrumentIds.size())
+                .collect(TreeMap::new,
+                        (map, i) -> map.put(instrumentIds.get(i), weights.get(i)),
+                        Map::putAll);
+
+        // Generate request using CreateReferencePortfolioRequest model
+        CreateReferencePortfolioRequest createRequest = new CreateReferencePortfolioRequest()
+                .code(F40PortfolioCode)
+                .displayName(F40PortfolioName)
+                .created(effectiveDate);
+
+        // Make the request using the reference portfolio API
+        referencePortfoliosApi.createReferencePortfolio(TestDataUtilities.TutorialScope, createRequest);
+
+        List<ReferencePortfolioConstituentRequest> constituentRequests = new ArrayList<>();
+
+        // Prepare individual constituent requests
+        for (String instrumentId : instrumentWeights.keySet()) {
+            TreeMap<String, String> instrumentIdentifier = new TreeMap<>();
+            instrumentIdentifier.put(TestDataUtilities.LUSID_INSTRUMENT_IDENTIFIER, instrumentId);
+
+            constituentRequests.add(new ReferencePortfolioConstituentRequest()
+                    .instrumentIdentifiers(instrumentIdentifier)
+                    .weight(instrumentWeights.get(instrumentId))
+                    .currency("GBP")
+            );
+        }
+
+        // Prepare bulk constituent upsert request
+        UpsertReferencePortfolioConstituentsRequest upsertRequest = new UpsertReferencePortfolioConstituentsRequest()
+                .constituents(constituentRequests)
+                .weightType(UpsertReferencePortfolioConstituentsRequest.WeightTypeEnum.STATIC)
+                .effectiveFrom(effectiveDate.toString());
+
+        // Make the request
+        referencePortfoliosApi.upsertReferencePortfolioConstituents(TestDataUtilities.TutorialScope,
+                F40PortfolioCode,
+                upsertRequest);
+
+        // Get constituents of portfolio
+        List<ReferencePortfolioConstituent> constituents = referencePortfoliosApi.getReferencePortfolioConstituents(TestDataUtilities.TutorialScope,
+                F40PortfolioCode,
+                null,
+                null,
+                null).getConstituents();
+
+        // Sorting constituents by their LUIDs so we can compare to the request LUIDs
+        constituents.sort(Comparator.comparing(ReferencePortfolioConstituent::getInstrumentUid));
+
+        // Validate number of constituents matches what was inserted
+        assertEquals(constituents.size(), instrumentIds.size());
+
```

Comment:
As your basically asserting for equality of instruments to weights and that they're mapped correctly can we simplify the below assertions. 

From you returned list of constituents can you construct another map of instrument to weights. Something like:

`Map<String,Double> portfolioConstituentsAndWeights = constituents.stream().collect(Collectors.toMap(
                ReferencePortfolioConstituent::getInstrumentUid,
                ReferencePortfolioConstituent::getWeight
        ));`

And then just assert that map exactly matches your original values in `instrumentWeights`?

Reply:
Would allow you to remove all the sorting as well in both the setup method and returned result as we don't really care about the orders in the map just that the instruments are present and mapped to the correct weights